presto-hive/src/main/java/com/facebook/presto/hive/s3/UnrecoverableS3OperationException.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.s3;
+
+import org.apache.hadoop.fs.Path;
+
+import static java.lang.String.format;
+
+/**
+ * This exception is for stopping retries for S3 calls that shouldn't be retried.
+ * For example, "Caused by: com.amazonaws.services.s3.model.AmazonS3Exception: Forbidden (Service: Amazon S3; Status Code: 403 ..."
+ */
+public class UnrecoverableS3OperationException
```

Comment:
This class is an implementation detail of `PrestoS3FileSystem`, used to stop retries on permanent errors. Given that the usages are different (constructors are different), I'd rather keep them separate, so please add a private copy to `S3SelectLineRecordReader`.

Reply:
I received feedback on earlier iterations to extract the exception in it's separate class because it is used to stop retries on permanent errors encountered in S3SelectLineRecordReader, as well. Both constructors are logically similar, as both try to provide the path that resulted in failed S3 request. 

With the above context, do you still believe we should split this?

Reply:
I'd still like to keep this separate. However, if you feel strongly about it, please do the refactor as a separate commit (before the addition of the S3 Select code).

Reply:
I don't feel strongly about this. Just wanted to get your feedback on previous discussions we had for this class. I will make the change in the next revision.

Reply:
Refactored